main.py

```diff
@@ -60,12 +67,18 @@ def __init__(self):
             GameState.PAUSE: self.pause_menu,
             GameState.SETTINGS: self.settings_menu,
             GameState.SHOP: self.shop_menu,
-            GameState.LEVEL: self.level
+            GameState.LEVEL: self.level,
+            GameState.INVENTORY: self.inventory_menu
         }
         self.current_state = GameState.MAIN_MENU
 
     def switch_state(self, state):
         self.current_state = state
+        if self.current_state == GameState.SAVE_AND_RESUME:
+            self.level.player.save()
+            self.current_state = GameState.LEVEL
+        if self.current_state == GameState.INVENTORY:
+            self.inventory_menu.refresh_buttons_content()
```

Comment:
so you let it as it is ?

Reply:
I what?

Reply:
we need to be consistent, I think we should either put everything in switch state or put everything in the action methods of the menus

Reply:
I don't know about that